scene/2d/tile_map_layer.cpp

```diff
@@ -1195,15 +1195,15 @@ void TileMapLayer::_navigation_draw_cell_debug(const RID &p_canvas_item, const V
 
 void TileMapLayer::_scenes_update(bool p_force_cleanup) {
 	// Check if we should cleanup everything.
-	bool forced_cleanup = p_force_cleanup || !enabled || !is_inside_tree() || tile_set.is_null();
```

Comment:
Disabling a layer should still remove the nodes I think.

Reply:
Hm, should disabling a layer remove the nodes? Or just disable them?

I'm imagining a scenario where you disable/re-enable a layer. You'd probably want your scene nodes to not get freed/re-instantiated.

Reply:
Disabling a layer makes it invisible and non-interactable, so the same should apply to nodes. If you want to achieve that while keeping the nodes' state, the solution could be temporarily removing them from scene tree and re-adding once the layer is enabled. It sounds complex to implement though.

Reply:
Hm, not only that, but it also seems like one should still be able to find/access/modify the nodes in the Layer subtree even if the Layer is disabled.

This is a more impactful change, but one solution might be:
1. Parent tile scene nodes to the Layer instead of the TileMap
2. Now, I _think_ setting Node.ProcessMode and CanvasItem.Visible on the Layer cover the exact functionality that Layer.enabled does

Reply:
CC @groud WDYT?

Reply:
Maybe we could go with removing for now, just to fix the regression, and improve it in a follow-up. Changing behavior of layer enabling wouldn't count as bugfix.